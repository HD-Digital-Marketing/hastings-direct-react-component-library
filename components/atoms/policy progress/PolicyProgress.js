import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './policy-progress.scss'

const PolicyProgress = props => {

    const {
        progress = 0, // Out of 120 (12 months)
        startDate = '',
        endDate = '',
        expired = false,
        cancelled = false,
        willStart = false,
        className = ''
    } = props

    const progressPolicyClass = classNames('a-policy-progress', className, {
        'a-policy-progress--error': expired || cancelled
    })

    let progressContent;
    let progressValue;

    if (progress && !willStart) {
        progressValue = progress
    } else if (expired || cancelled) {
        progressValue = 120
    } else {
        progressValue = 0
    }

    if (expired) {
        progressContent = <div className="a-policy-progress__note">This policy expired on {endDate}</div>
    } else if (cancelled) {
        progressContent = <div className="a-policy-progress__note">This policy was cancelled on {endDate}</div>
    } else if (willStart) {
        progressContent = <div className="a-policy-progress__note">This policy will start on {startDate}</div>
    } else {
        progressContent = <div className="a-policy-progress__note"><span className="a-policy-progress__start">{startDate}</span><span className="a-policy-progress__end">{startDate}</span></div>
    }

    return(
        <div className={progressPolicyClass}>
            <progress className="a-policy-progress__bar" value={progressValue} max="120">{progress}</progress>
            {progressContent}
        </div>
    )
}

PolicyProgress.propTypes = {
    progress: PropTypes.number,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    expired: PropTypes.bool,
    cancelled: PropTypes.bool,
    willStart: PropTypes.bool,
    className: PropTypes.string
}

export default PolicyProgress
