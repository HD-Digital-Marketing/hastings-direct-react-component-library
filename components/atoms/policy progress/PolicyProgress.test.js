import React from 'react';
import renderer from 'react-test-renderer';
import PolicyProgress from './PolicyProgress'

describe('Check the progress bar renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <PolicyProgress progress={70} startDate="07 03 2020" endDate="08 03 2021"/>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})