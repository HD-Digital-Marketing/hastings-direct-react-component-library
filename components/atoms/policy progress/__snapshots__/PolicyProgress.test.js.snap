// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the progress bar renders it should match the snapshot 1`] = `
<div
  className="a-policy-progress"
>
  <progress
    className="a-policy-progress__bar"
    max="120"
    value={70}
  >
    70
  </progress>
  <div
    className="a-policy-progress__note"
  >
    <span
      className="a-policy-progress__start"
    >
      07 03 2020
    </span>
    <span
      className="a-policy-progress__end"
    >
      07 03 2020
    </span>
  </div>
</div>
`;
